From dd7575ba44f0e255e3d3c04bc60648a8d41a18d4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Tue, 18 May 2021 15:05:53 +0300
Subject: [PATCH] [LLD] [MinGW] Pass the canExitEarly parameter through
 properly

The MinGW driver passed a hardcoded true to this parameter
since 6f4e255219f2a7878d3, but when the MinGW driver got the
canExitEarly parameter for consistency in b11386f9be9b2dc7276, this
call was missed so it wasn't passed on properly.
---
 MinGW/Driver.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/MinGW/Driver.cpp b/MinGW/Driver.cpp
index e6c6efdd2b95..86a898de7517 100644
--- a/MinGW/Driver.cpp
+++ b/MinGW/Driver.cpp
@@ -437,5 +437,5 @@ bool mingw::link(ArrayRef<const char *> argsArr, bool canExitEarly,
   std::vector<const char *> vec;
   for (const std::string &s : linkArgs)
     vec.push_back(s.c_str());
-  return coff::link(vec, true, stdoutOS, stderrOS);
+  return coff::link(vec, canExitEarly, stdoutOS, stderrOS);
 }
-- 
2.31.1.windows.1

